src/onegov/election_day/collections/archived_results.py

```diff
@@ -259,3 +262,210 @@ def delete(self, item, request):
+            )
+        )
+
+        return items
+
+    @staticmethod
+    def term_to_tsquery_string(term):
+        """ Returns the current search term transformed to use within
+        Postgres ``to_tsquery`` function.
+        Removes all unwanted characters, replaces prefix matching, joins
+        word together using FOLLOWED BY.
+        """
+
+        def cleanup(word, whitelist_chars=',.-_'):
+            result = ''.join(
+                (c for c in word if c.isalnum() or c in whitelist_chars)
+            )
+            return f'{result}:*' if word.endswith('*') else result
+
+        parts = (cleanup(part) for part in (term or '').split())
+        return ' <-> '.join(tuple(part for part in parts if part))
+
+    @staticmethod
+    def match_term(column, language, term):
+        """ Usage:
+         model.filter(match_term(model.col, 'german', 'my search term')) """
+        document_tsvector = func.to_tsvector(language, column)
+        ts_query_object = func.to_tsquery(language, term)
+        return document_tsvector.op('@@')(ts_query_object)
+
+    @staticmethod
+    def filter_text_by_locale(column, term, locale=None):
+        """ Returns an SqlAlchemy filter statement based on the search term.
+        If no locale is provided, it will use english as language.
+        ``to_tsquery`` creates a tsquery value from term, which must consist of
+         single tokens separated by the Boolean operators
+        & (AND), | (OR) and ! (NOT).
+        ``to_tsvector`` parses a textual document into tokens, reduces the
+        tokens to lexemes, and returns a tsvector which lists the lexemes
+        together with their positions in the document. The document is
+        processed according to the specified or default text search
+        configuration. """
+
+        mapping = {'de_CH': 'german', 'fr_CH': 'french', 'it_CH': 'italian',
+                   'rm_CH': 'english'}
+        return SearchableArchivedResultCollection.match_term(
+            column, mapping.get(locale, 'english'), term
+        )
+
+    @property
+    def term_filter(self):
+        assert self.term
+        assert self.locale
+        term = SearchableArchivedResultCollection.term_to_tsquery_string(
+            self.term)
+        # The title is a translations hybrid, .title is a shorthand
+        return (
+            SearchableArchivedResultCollection.filter_text_by_locale(
+                ArchivedResult.shortcode, term, self.locale),
+            SearchableArchivedResultCollection.filter_text_by_locale(
+                ArchivedResult.title, term, self.locale)
+        )
+
+    def check_from_date_to_date(self):
+        if not self.to_date or not self.from_date:
+            return
+
+        if self.to_date > date.today():
+            self.to_date = date.today()
+        if self.from_date > self.to_date:
+            self.from_date = self.to_date
+
+    def query(self, no_filter=False, sort=True):
+        if no_filter:
```

Comment:
This block of code could use some spacing. You seem to have some guard clauses, checks and then default values. I would say at least the guard clauses/checks should be visibly separated from the defaults with an extra empty line.